tests/src/python/test_qgsvectorlayertemporalproperties.py

```diff
@@ -273,6 +275,22 @@ def testSingleFieldMode(self):
         # OR start of feature < end of range AND start of feature > start of range - duration
         self.assertEqual(props.createFilterString(context, range), '("start_field" > make_datetime(2019,3,1,11,12,13) AND "start_field" < make_datetime(2020,5,6,8,9,10)) OR "start_field" IS NULL')
 
+        # since 3.22 there is also the option to include the end of the feature event
+        props.setLimitMode(QgsVectorLayerTemporalProperties.LimitMode.ModeIncludeBeginIncludeEnd)
+        # map range              [-------------------------)
+        # feature ranges         .                         . [-------]  (false)
+        #                        .                         [-------]    (false)
+        #                        .                     [---.---]        (true)
+        #                        .            [-------]    .            (true)
+        #                        [-------]                 .            (true)
+        #                    [---.---]                     .            (true)
+        #                [-------]                         .            (true)
+        #          [-------]     .                         .            (false)
+        # => start of feature < end of range AND start of feature + duration > start of range
```

Comment:
Looking at the diagram, isn't this 

```suggestion
        # => start of feature < end of range AND start of feature + duration >= start of range
```
?